sklearn/preprocessing/data.py

```diff
@@ -1333,7 +1333,8 @@ def binarize(X, threshold=0.0, copy=True):
     using the ``Transformer`` API (e.g. as part of a preprocessing
     :class:`sklearn.pipeline.Pipeline`)
     """
-    X = check_array(X, accept_sparse=['csr', 'csc'], copy=copy)
+    X = check_array(X, accept_sparse=['csr', 'csc'], copy=copy,
```

Comment:
This raised no errors... This was done just to fix the input dtype to float (preemptively)... Is this unnecessary?